src/check_zfs.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+"""InnoGames Monitoring Plugins - ZFS pool status check
+
+Copyright (c) 2020 InnoGames GmbH
+"""
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from subprocess import check_output, STDOUT, CalledProcessError
+from argparse import ArgumentParser
+from sys import exit
+import re
+import platform
+
+
+# Nagios plugin exit codes
+class ExitCodes:
+    ok = 0
+    warning = 1
+    critical = 2
+    unknown = 3
+
+
+def main():
+    parser = ArgumentParser(
+        description=('Check if all ZFS pools are online')
+    )
+    
+    args = parser.parse_args()
+
+    try:
+        proc = check_output(
+            ['/sbin/zpool', 'list', '-H'],stderr=STDOUT,
+        ).decode()
+    except OSError as e:
+        print('UNKNOWN: can\'t read zpool status: {}'.format(e))
+        return ExitCodes.unknown
+    except CalledProcessError as e:
+        print('UNKNOWN: can\'t read zpool status: {}'.format(e.output))
+        return ExitCodes.unknown
+    
+    exit_multiline = []
+    lines = proc.splitlines()
+
+    if len(lines):
+        exit_code = ExitCodes.ok
+        exit_header = 'All zpools are fine'
+        for line in lines:
+            zpool_code, zpool_name, zpool_state = parse_line(line)
+            exit_multiline.append('{}: {}'.format(zpool_name, zpool_state))
+            if zpool_code > exit_code:
+                exit_code = zpool_code
+                exit_header = 'Some zpools are not online!'
+    else:
+        exit_code = ExitCodes.unknown
+        exit_header = 'No zpools found!'
+
+
+    print(exit_header)
+    if exit_multiline:
+        print('\n'.join(exit_multiline))
+    return exit_code
+
+def parse_line(line):
+    """
+    The output looks like this, with -H there are real tabs separating fields:
+
+    # sudo zpool list -H | tr '\t' _
+    tank_130T_75.9T_54.1T_-_22%_58%_1.00x_ONLINE_-
+    ...
+    """
+
+    line_split = line.split('\t')
```

Comment:
It could also possible like that (if it fits with the line limit):
```suggestion
    name, _, _, _, _, _, _, _, code = line.split('\t')
```
`name` would hold `line_split[0]` then and `code` `line_split[8]`. `_` would hold the last occurrence, so `line_split[7]`

Both is fine for me here, you can choose.